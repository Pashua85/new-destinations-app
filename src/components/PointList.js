import React from 'react';
import Reorder from 'react-reorder';
import PointListItem from './PointListItem';

export default class PointList extends React.Component {

  itemClicked = (event, clickedItem, itemsIndex) => {
    if(event.target.closest('.remove-btn')) {
      this.props.deletePoint(clickedItem.pointId);
    }
  };

  onReorder =  (
    event,
    itemThatHasBeenMoved, 
    itemsPreviousIndex, 
    itemsNewIndex, 
    reorderedArray) => {
    this.props.handleReorder(reorderedArray);
  };

  render() {

    return (
      <Reorder
        itemKey='pointId'
        lock='horizontal'
        holdTime='100'
        list={this.props.points}
        template={PointListItem}
        callback={this.onReorder}
        itemClicked={this.itemClicked}
        itemClass={'list-item'}
      />
    )
  }
};