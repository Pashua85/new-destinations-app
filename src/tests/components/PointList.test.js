import React from 'react';
import { shallow } from 'enzyme';
import PointList from '../../components/PointList';
import points from '../utils/points';

let wrapper, handleReorder;

beforeEach(() => {
  handleReorder = jest.fn();
  wrapper = shallow(
    <PointList 
      points={points}
      handleReorder={handleReorder}
    />
  );  
});

test('should render PointList with points', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should call handleReorder on changing order', () => { 
  wrapper.find('Reorder').prop('callback')();
  expect(handleReorder).toHaveBeenCalled();
});
