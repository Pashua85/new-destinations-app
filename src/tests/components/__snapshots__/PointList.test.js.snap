// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PointList with points 1`] = `
<Reorder
  callback={[Function]}
  holdTime="100"
  itemClass="list-item"
  itemClicked={[Function]}
  itemKey="pointId"
  list={
    Array [
      Object {
        "latLng": Object {
          "lat": 56.80950300000001,
          "lng": 60.606506999999965,
        },
        "pointId": "ChIJc8sXNttuwUMRWSTSqEBCxrY",
        "pointName": "Point one",
      },
      Object {
        "latLng": Object {
          "lat": 56.8073729,
          "lng": 60.62031009999998,
        },
        "pointId": "ChIJASguT89uwUMRMbx0H8V2uNk",
        "pointName": "Point two",
      },
      Object {
        "latLng": Object {
          "lat": 56.854948,
          "lng": 60.661017000000015,
        },
        "pointId": "ChIJ6TiVJjVswUMRRi8Evo28BM4",
        "pointName": "Point three",
      },
    ]
  }
  lock="horizontal"
  template={[Function]}
/>
`;
